src/lib/reasoners/enum_rel.ml

```diff
@@ -156,6 +189,14 @@ let add_diseq hss sm1 sm2 dep env eqs =
 
   |  _ -> env, eqs
 
+(* Update the domains of the semantic values [sm1] and [sm2] according to
+   the equation [sm1 = sm2]. More precisely, we replace their domains by
+   the intersection of these domains.
+
+   In any case, we produce an equality if the domain of these semantic values
+   becomes a singleton.
+
+   @raise Ex.Inconsistent if the domains of [sm1] and [sm2] are disjoint. *)
```

Comment:
I know this comment is a bit wrong as the function doesn't produce a new equality when one the semantic value is a constructor. I don't know why? Maybe we don't need to it here because it will be done somewhere else.

Reply:
If we execute this function with `Alien _, Cons _` or `Cons _, Alien _` we are already processing the equality that we would be producing :)